framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+    << "GetObject for an untracked, invalid bundle should return nullopt";
+
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackedWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  auto tracked = bundleTracker.GetTracked();
+  EXPECT_EQ(0, tracked.size())
+    << "GetTracked() should return an empty map before Open()";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundleMain = cppmicroservices::testing::GetBundle("main", context);
+  Bundle bundleSys =
+    cppmicroservices::testing::GetBundle("system_bundle", context);
+  Bundle bundleA = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  tracked = bundleTracker.GetTracked();
+
+  bool mainTracked = tracked.find(bundleMain) != tracked.end();
+  bool sysTracked = tracked.find(bundleSys) != tracked.end();
+  bool bundleATracked = tracked.find(bundleA) != tracked.end();
+
+  EXPECT_TRUE(mainTracked) << "GetTracked() should include the main bundle";
+  EXPECT_TRUE(sysTracked) << "GetTracked() should include the system bundle";
+  EXPECT_TRUE(bundleATracked) << "GetTracked() should include the test bundle";
+
+  bundleTracker.Close();
+  tracked = bundleTracker.GetTracked();
+  EXPECT_EQ(0, tracked.size())
+    << "GetTracked() should return an empty map after Close()";
+}
+
+TEST_F(BundleTrackerMethodTest, IsEmptyWorks)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  EXPECT_TRUE(bundleTracker.IsEmpty())
+    << "Unopened BundleTracker should be empty";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(bundle, bundleTracker.GetObject(bundle))
+    << "BundleTracker failed to track a bundle";
+  EXPECT_FALSE(bundleTracker.IsEmpty())
+    << "BundleTracker should not be empty when a bundle is tracked";
+
+  bundleTracker.Close();
+  EXPECT_TRUE(bundleTracker.IsEmpty())
+    << "Closed BundleTracker should be empty";
+}
+
+TEST_F(BundleTrackerMethodTest, SizeWorks)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_UNINSTALLED);
+  BundleTracker<> bundleTracker(context, stateMask);
+  EXPECT_EQ(0, bundleTracker.Size())
+    << "Size of unopened BundleTracker was not 0";
+
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  EXPECT_EQ(0, bundleTracker.Size())
+    << "Size should be 0 when no bundles has entered a tracked state";
+
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  bundle.Uninstall();
+  EXPECT_EQ(1, bundleTracker.Size())
+    << "Size should be 1 when 1 bundle entered a tracked state";
+
+  bundleTracker.Close();
+  EXPECT_EQ(0, bundleTracker.Size())
+    << "Size of closed BundleTracker was not 0";
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksAfterBundleAdded)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.